tests/rdma/rdma_env.py

```diff
@@ -0,0 +1,151 @@
+            print("Neither kernel builtin nor out-of-tree rdma_rxe.ko found. Abort")
+            print("Please run the following commands to build out-of-tree RXE on Linux-6.5, then retry:")
+            print("\t~# mkdir -p " + rxedir)
+            print("\t~# git clone https://github.com/pizhenwei/rxe.git " + rxedir)
+            print("\t~# cd " + rxedir)
+            print("\t~# make")
+            os._exit(1);
+
+        cmd = "insmod " + rxekmod
+        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        if p.wait():
+            os._exit(1);
+
+    print("Valkey Over RDMA install RXE [OK]")
+
+    softrdma = "rxe_" + interface
+    cmd = "rdma link add " + softrdma + " type rxe netdev " + interface
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    if p.wait():
+        outs, _ = p.communicate()
+        print("Valkey Over RDMA install RXE [FAILED]")
+        print("---------------\n" + outs.decode() + "---------------\n")
+        os._exit(1);
+
+    print("Valkey Over RDMA add RXE device <%s> [OK]" % softrdma)
+
+
+# find any IPv4 available networking interface
+def find_iface():
+    interfaces = netifaces.interfaces()
+    for interface in interfaces:
+        if interface == "lo":
+            continue
+
+        addrs = netifaces.ifaddresses(interface)
+        if netifaces.AF_INET not in addrs:
+            continue
+
+        return interface
+
+
+def setup_rdma(driver, interface):
+    if interface == None:
+        interface = find_iface()
+
+    prepare_ib()
+    if driver == "rxe":
+        prepare_rxe(interface)
+    else:
+        print("rxe is currently supported only")
+        os._exit(1);
+
+
+# iterate /sys/class/infiniband, find any all virtual RDMA device, and remove them
+def cleanup_rdma():
+    # Ex, /sys/class/infiniband/mlx5_0
+    # Ex, /sys/class/infiniband/rxe_eth0
+    # Ex, /sys/class/infiniband/siw_eth0
+    ibclass = "/sys/class/infiniband/"
+    try:
+        for dev in os.listdir(ibclass):
+            # Ex, /sys/class/infiniband/rxe_eth0/ports/1/gid_attrs/ndevs/0
+            origpath = os.readlink(ibclass + dev)
+            if "virtual" in origpath:
+                subprocess.Popen("rdma link del " + dev, shell=True).wait()
+                print("Remove virtual RDMA device : " + dev + " [OK]")
+    except os.error:
+        return None
+
+    # try to remove RXE driver from kernel, ignore error
+    subprocess.Popen("rmmod rdma_rxe 2> /dev/null", shell=True).wait()
+
+    # try to remove SIW driver from kernel, ignore error
+    subprocess.Popen("rmmod rdma_siw 2> /dev/null", shell=True).wait()
+
+    return None
+
+
+if __name__ == "__main__":
+    examples = """examples:
+    ./rdma_env.py -o [setup|cleanup] # setup or cleanup soft RDMA environment
+    ./rdma_env.py -d [rxe|siw]       # specify soft RDMA driver, rxe by default
+    ./rdma_env.py -i [IFACE]         # network interface, auto-select any available interface by default
```

Comment:
These don't look like examples. They are documentation of the options. `argparse` can already print this with `--help`.

If you want examples, maybe combine this into one real example and use realistic values. (Or we don't need an example.)

Reply:
Right, remove these examples in the next version. Thanks!